modules/ray_tracing/test/src/userobjects/SingleTraceLineSourceTest.C

```diff
@@ -0,0 +1,50 @@
+#include "SingleTraceLineSourceTest.h"
+#include "NonlinearSystemBase.h"
+#include "SystemBase.h"
+
+registerMooseObject("RayTracingTestApp", SingleTraceLineSourceTest);
+
+InputParameters
+SingleTraceLineSourceTest::validParams()
+{
+  auto params = RepeatableRayStudy::validParams();
+  params.addParam<TagName>("residual_vector_tag",
+                           "the vector tag for the residual tag you will accumulate into");
+  params.set<bool>("allow_other_flags_with_prekernels") = true;
+  return params;
+}
+
+SingleTraceLineSourceTest::SingleTraceLineSourceTest(const InputParameters & parameters)
+  : RepeatableRayStudy(parameters),
+    _residual_tag_name(getParam<TagName>("residual_vector_tag")),
+    _has_traced(false)
+{
+}
+
+void
+SingleTraceLineSourceTest::execute()
+{
+  if (_current_execute_flag == EXEC_TIMESTEP_BEGIN)
+  {
+    _has_traced = false;
+    return;
+  }
+
+  if (_fe_problem.currentlyComputingJacobian())
```

Comment:
Maybe just exclude `EXEC_NONLINEAR` from the execution flags?

Reply:
It looks like this works. But would you be able to tell me about what the difference between these two implementations is?

Reply:
Essentially my suggestion doesn't create the condition that you need to fix